kbase-extension/static/kbase/js/common/cellComponents/fieldTableCellWidget.js

```diff
@@ -63,90 +55,106 @@ define([
+                                },
+                            }),
+                        },
+                        [spec.ui.label || spec.ui.id]
+                    ),
+                    div({
+                        class: `${cssBaseClass}__input_control`,
+                        id: ids.inputControl,
+                        dataElement: 'input-control',
+                    }),
+                ]
+            );
 
             return {
                 content: content,
-                places: ids
+                places: ids,
             };
         }
 
         // LIFECYCLE
 
         function attach(node) {
-            return Promise.try(function() {
-                parent = node;
-                let containerDiv = document.createElement('div');
-                containerDiv.classList.add(`${cssBaseClass}__param_container`);
-
-                container = parent.appendChild(containerDiv);
-                var events = Events.make({
-                    node: container
-                });
+            parent = node;
+            let containerDiv = document.createElement('div');
+            containerDiv.classList.add(`${cssBaseClass}__param_container`);
 
-                var rendered = render(events);
-                container.innerHTML = rendered.content;
-                events.attachEvents();
-                // TODO: use the pattern in which the render returns an object,
-                // which includes events and other functions to be run after
-                // content is added to the dom.
-                PR.prettyPrint(null, container);
-
-                places = {
-                    field: document.getElementById(fieldId),
-                    message: document.getElementById(rendered.places.message),
-                    inputControl: document.getElementById(rendered.places.inputControl)
-                };
-                if (inputControl.attach) {
-                    return inputControl.attach(places.inputControl);
-                }
+            container = parent.appendChild(containerDiv);
+            var events = Events.make({
+                node: container,
             });
+
+            var rendered = render(events);
+            container.innerHTML = rendered.content;
+            events.attachEvents();
+            // TODO: use the pattern in which the render returns an object,
+            // which includes events and other functions to be run after
+            // content is added to the dom.
+            PR.prettyPrint(null, container);
+
+            places = {
+                field: document.getElementById(fieldId),
+                message: document.getElementById(rendered.places.message),
+                inputControl: document.getElementById(rendered.places.inputControl),
+            };
+
+            //TODO: why do this? it seems recursive? could we even call attach in this case? it's not a public method
+            // if (inputControl.attach) {
```

Comment:
does anyone know why this is here? i commented it out because i couldn't figure out what it was meant to achieve

Reply:
If the commented text isn't needed, could you remove it?